### DISCUSSION TITLE
[Feature Request] LoopAuditioneer Waveform Overlay Window

### DISCUSSION THREAD

**Discussion by vpoguru** [GrandOrgue/LoopAuditioneer#7] [Ideas]
1. It would be great if the waveform overlay window had buttons for the previous and next sample. Currently, if you are reviewing a complete rank, you need to do the following to get to the next sample:

- Close the waveform overlay window.
- Double click the next sample.
- Open the waveform overlay window again.

With the ‘Previous’ and ‘Next’ buttons, this would be a one-click solution. If you are considering this function, you should insert the name of the currently open file in the title bar or before "Loop #: n" to ensure that the user knows which sample he is looking at.

2. It would be great if the waveform overlay window had a play button. This could be a single button with a play icon when the sound is not playing and turn into a stop button when the user wants to stop the sound.

I like the "Transport > Loop only" option. If checking long loops, this can be still time consuming, so it would be great if the playback started 1 or 2 seconds before the jump dwEnd to dwStart and then jumped back to the position 1 or 2 seconds before dwEnd after 1 to 2 seconds. If you crossfade the 'inner' jump, the listener can concentrate on the jump dwEnd to dwStart.

3. I would love it if LoopAuditioneer would save the position and size of the waveform overlay window.

**Comment by larspalo** [comment]
I basically agree with all the main points above, but I think that I'll go about fixing it in a slightly different way.

My idea is to make the waveform overlay into a modeless window that will more take the functionality of waveform magnification. As modeless it would react to whatever loop or cue is currently selected in the active file and change its display/appearance accordingly (waveform overlay when a loop is selected and just a waveform magnification when a cue is selected). Also, that would allow playback as the main window is still usable and they will update according to changes both ways.

Saving the modeless window position should be possible, however it seems that Wayland on Ubuntu 24.04 has some issues with restoring that exact position.

> I like the "Transport > Loop only" option. If checking long loops, this can be still time consuming, so it would be great if the playback started 1 or 2 seconds before the jump dwEnd to dwStart and then jumped back to the position 1 or 2 seconds before dwEnd after 1 to 2 seconds. If you crossfade the 'inner' jump, the listener can concentrate on the jump dwEnd to dwStart.

Personally, what I do when I'm in a hurry is to just use the spacebar repeatedly to play and stop the sample quickly after the loop point to avoid having to listen to the whole loop every time. High listening volume and boost also helps identifying issues but it quickly becomes very tiresome for the ears...

Cutting away playback of the middle part of a loop might hide defects that exist and perhaps should be avoided in the loops, so I'm not so fond of that.
